pallets/cdd-offchain-worker/src/lib.rs

```diff
@@ -216,49 +253,40 @@ impl<T: Trait> frame_support::unsigned::ValidateUnsigned for Module<T> {
             let last_unsigned_at = <LastExtSubmittedAt<T>>::get();
             ensure!(
-                last_unsigned_at + T::CoolingInterval::get() <= *block_number,
+                last_unsigned_at + T::CoolingInterval::get() <= payload.block_number,
                 InvalidTransaction::Stale
             );
 
             // Let's make sure to reject transactions from the future.
             ensure!(
-                &<system::Module<T>>::block_number() >= block_number,
+                &<system::Module<T>>::block_number() >= &payload.block_number,
                 InvalidTransaction::Future
             );
 
-            // Get the key from the store to verify whether the signed signature is
-            // generated from the validator or not.
-            let signer_id = T::SignerId::all()
-                .into_iter()
-                .next()
-                .ok_or(InvalidTransaction::BadProof)?;
-
-            let signature_valid = target.using_encoded(|encoded_nominators| {
-                signer_id.verify(&encoded_nominators, &signature)
-            });
             // validating signature
-            ensure!(signature_valid, InvalidTransaction::BadProof);
-
-            Ok(ValidTransaction {
-                // We set the priority to the max value - 10000. ~ near to high priority
-                // TODO: We can change after discussion
-                priority: TransactionPriority::max_value() - 10000_u64,
-                requires: vec![],
-                // We set the `provides` tag to be the same as `last_unsigned_at`. This makes
-                // sure only one transaction produced after `last_unsigned_at` will ever
-                // get to the transaction pool and will end up in the block.
-                // We can still have multiple transactions compete for the same "spot",
-                // and the one with higher priority will replace other one in the pool.
-                provides: vec![codec::Encode::encode(&(KEY_TYPE.0, last_unsigned_at))],
-                // The transaction is only valid for next n blocks. After that it's
-                // going to be revalidated by the pool. n = cooling interval
-                longevity: (T::CoolingInterval::get()).saturated_into::<u64>(),
-                propagate: true,
-            })
+            ensure!(
+                SignedPayload::<T>::verify::<T::AuthorityId>(payload, signature.clone()),
+                InvalidTransaction::BadProof
+            );
+
+            ValidTransaction::with_tag_prefix("CddOffchainWorker")
+                .priority(T::UnsignedPriority::get().saturating_add(
+                    1000 * u64::try_from(payload.nominators.len()).unwrap_or_default(),
+                ))
+                // The transaction is only valid for next 5 blocks. After that it's
+                // going to be revalidated by the pool.
+                .longevity(5)
+                // Not to be propagated to peer nodes.
+                .propagate(false)
```

Comment:
Why is propagation not allowed?

Reply:
I don't want other nodes to create the transactions, I think other than authored node.

Reply:
That's not how this will work. All nodes will create the tx regardless of propagation. However, this will be a wasted effort since they won't be able to broadcast their work if they are not author.

Reply:
I think by default only authoring nodes will run off-chain workers:
```
        --offchain-worker <ENABLED>
            Should execute offchain workers on every block.
            
            By default it's only enabled for nodes that are authoring new blocks. [default: WhenValidating]  [possible
            values: Always, Never, WhenValidating]
```

Reply:
Yes, from "all nodes", I meant all nodes who are validators. The problem here is that only the node that is authoring the current node will be able to include the transaction if it is not being propagated. Other validator nodes will just be wasting resources.

Reply:
Is there an easy way to only have the block authoring node run an off-chain worker? In practice these off-chain workers can run every era - so the amount of work should be small.

Reply:
> Is there an easy way to only have the block authoring node run an off-chain worker? 

Offchain worker and block authoring are independent, asynchronous processes. We can not tie them together. An offchain worker may take multiple blocks of time to execute.